src/diffusers/loaders.py

```diff
@@ -946,11 +946,15 @@ def _modify_text_encoder(self, attn_processors: Dict[str, LoRAAttnProcessor]):
                 lora_layer = getattr(attn_processors[name], self._get_lora_layer_attribute(name))
                 old_forward = module.forward
 
-                def new_forward(x):
-                    return old_forward(x) + lora_layer(x)
+                # create a new scope that locks in the old_forward, lora_layer value for each new_forward function
+                def make_new_forward(old_forward, lora_layer):
+                    def new_forward(x):
+                        return old_forward(x) + lora_layer(x)
```

Comment:
Small comment: if you load lora with e.g. `pipeline.load_lora_weights("experiments/base_experiment")` more than once, then this monkey patch becomes recursive!

Reply:
Thanks for your suggestion! Indeed, it seems that might be the case. I wonder if it might be better to create a mechanism to remove the moneky-patch. @sayakpaul WDYT?

Reply:
Best is indeed to do something else than monkey-patching, but a flag like `override_forward=False` as an arg would also be helpful to disable the monkey-patching.

Reply:
We're currently discussing this internally, and will keep y'all posted.

Reply:
Meanwhile, 

> Best is indeed to do something else than monkey-patching, but a flag like override_forward=False as an arg would also be helpful to disable the monkey-patching.

@rvorias could you elaborate what you mean here?

Reply:
Have `load_lora_weights` parse a kwarg `override_text_encoder_forward:bool`, then pass that to `_modify_text_encoder`. Then condition line 957 on this flag.

This is useful in contexts where you want to load arbitrary lora weights on the fly in a long-running SD inference engine.
Right now, calling `load_lora_weights` multiple times causes you to override the forward function multiple times and thus the lora addition term will get nested.

If you add the flag+condition you can still have the new lora weights to load, but you don't override the forward again and again.

Reply:
Would you be willing to open a PR for this? We're more than happy to help you with that :-)